app/components/subcase/description-panel/view.js

```diff
@@ -45,59 +42,54 @@ export default class SubcaseDescriptionView extends Component {
-        'filter[:has-no:next-version]': 't',
-        sort: '-created',
-      });
-      const agenda = yield agendaitem.agenda;
-      const meeting = yield agenda.createdFor;
-      yield meeting?.kind;
-      const decisionPublicationActivity = yield meeting
-        .belongsTo('internalDecisionPublicationActivity')
-        .reload();
-      yield decisionPublicationActivity?.status; // used in get-functions above
-      // load decisionActivity
-      // agenda-activities are propagated by yggdrail on agenda approval, treatments/decision-activities only when decisions are released
-      const treatment = yield agendaitem?.treatment;
-      const decisionActivity = yield treatment?.decisionActivity;
-      const resultCode = yield decisionActivity?.decisionResultCode;
-      // Other profiles should not have the latest decision when decisions have not been released yet
-      if (decisionActivity) {
-        // the last decision might be null, keep only the last one that exists
-        this.latestDecisionActivity = decisionActivity;
-      }
+    if (relatedAgendas.length === 0)
+      return;
+
+    // Check if we can access the last agenda using Ember Data, that means it
+    // has been propagated and we only care about the latest agenda
+    const lastRecord = relatedAgendas[0];
+    if (lastRecord.visible) {
+      // The latest meeting is visible to the current user
+      // We can just fetch it using the store and use the real records
+      const meeting = await this.store.findRecord('meeting', lastRecord.meeting.id);
+      const agenda = await this.store.findRecord('agenda', lastRecord.agenda.id);
+      const agendaitem = await this.store.findRecord('agendaitem', lastRecord.agendaitem.id);
+      this.latestMeetingModels = { meeting, agenda, agendaitem };
+    } else {
+      // The latest meeting is not visible, we should display it but not as a link
+      const visibleRecord = relatedAgendas.find((record) => record.visible);
 
-      this.modelsOfMeetings.push([meeting, agenda, agendaitem, resultCode]);
-      // we need this multiple times in the template and navigating the nested array each time is bothersome
-      if (index === agendaActivities.length - 1) {
-        this.latestMeeting = meeting;
+      // At least one meeting must be visible to the user to display
+      // If no meeting is yet propagated, we also don't show the unpropagated one
+      if (visibleRecord) {
+        const meeting = await this.store.findRecord('meeting', visibleRecord.meeting.id);
+        const agenda = await this.store.findRecord('agenda', visibleRecord.agenda.id);
+        const agendaitem = await this.store.findRecord('agendaitem', visibleRecord.agendaitem.id);
+
+        this.postponedMeetingModels =  {
+          meeting: lastRecord.meeting,
+          agenda: lastRecord.agenda,
+          agendaitem: lastRecord.agendaitem
+        };
+        this.latestMeetingModels = { meeting, agenda, agendaitem };
```

Comment:
Bit odd this one. The latestRecord is never visible on active agendas.
As in, Approved agenda A is propagated, design B is not.
Because we queried as sudo in the backend we are checking agenda B which we cannot see and therefore cannot make a link to.
But we can make a link to A, and we no longer do that.
With the postponing locally.
I have a working link for the old meeting, but no link for the new meeting even though atleast 1 agendaitem version of the agendaitems is approved

Showing 2 agendas, only 1 link
![image](https://github.com/user-attachments/assets/5d81ef44-9553-4308-ab5c-2bbf9fbe4991)

We can access the agenda with a link (A, not B)
![image](https://github.com/user-attachments/assets/0c436e00-13da-4fbf-a9eb-bd5c1fee8ab6)

Is it enough that we DO have a correct link lower on the page?

![image](https://github.com/user-attachments/assets/f0d2e6e1-b59d-4fe0-9c79-d7fb8485c36d)

Reply:
ow, and with profile "overheid" we are just not linking the second meeting even though it can see agenda A
AS "intern-overheid"
![image](https://github.com/user-attachments/assets/a2826fe3-2ab6-461a-9bfd-87bef3d8f7db)

![image](https://github.com/user-attachments/assets/50c41695-c79f-4ee6-a04c-6c6aade8fe70)

Reply:
fixed in service